miasm2/core/asmblock.py

```diff
@@ -359,132 +323,209 @@ class AsmSymbolPool(object):
     """
 
     def __init__(self):
-        self._labels = set()
-        self._name2label = {}
-        self._offset2label = {}
-        self._label_num = 0
+        self._loc_keys = set()
+
+        self._loc_key_to_offset = {}
+        self._loc_key_to_name = {}
+
+        self._name_to_loc_key = {}
+        self._offset_to_loc_key = {}
+
+        self._loc_key_num = 0
 
-    def add_label(self, name, offset=None):
+    def loc_key_to_offset(self, loc_key):
         """
-        Create and add a label to the symbol_pool
-        @name: label's name
-        @offset: (optional) label's offset
+        Return offset of @loc_key, None else.
+        @loc_key: LocKey instance
         """
-        label = AsmLabel(name, offset)
+        return self._loc_key_to_offset.get(loc_key, None)
+
+    def loc_key_to_name(self, loc_key):
+        """
+        Return name of @loc_key.
+        @loc_key: LocKey instance
+        """
+        return self._loc_key_to_name.get(loc_key)
+
+    def add_location(self, name, offset=None):
```

Comment:
`name` could be at the same "level" of `offset`, ie. also optional. If I understand correctly, for now it is:
`add_location(offset)` to add an offset.
The API would be more consistent with:
`add_location(offset=offset)` and `add_location(name=name)`, instead of having 2 separate behavior if the first argument is a int or not

Reply:
See with @commial: not a bug as name is mandatory